trunk/source/Library-Sifa/src/de/uni_freiburg/informatik/ultimate/lib/sifa/domain/IntervalDomain.java

```diff
@@ -209,81 +76,89 @@ private Optional<Map<TermVariable, Interval>> dnfDisjunctToIntervals(
 				if (!updatedLhsInterval.isPresent()) {
 					continue;
 				} else if (updatedLhsInterval.get().isBottom()) {
-					return Optional.empty();
+					return null;
 				}
 				updated = true;
 			}
 		}
 		if (updated) {
 			// maxIter limit reached
 			// TODO research whether this only happens if dnfDisjunct is unsat.
-			//      If so, then return bottom
+			// If so, then return bottom
 			mLogger.warn("Interval conversion did not stabilize in %d iterations. "
 					+ "Over-approximation may be very coarse.", maxIterations);
 			mLogger.debug("Relations used to update are %s.", solvedRelations);
 			mLogger.debug("Interval values after last iteration are %s.", varToInterval);
 		}
-		return Optional.of(Collections.unmodifiableMap(varToInterval));
+		return new NonrelationalState<>(varToInterval);
 	}
 
-	private List<SolvedBinaryRelation> dnfDisjunctToSolvedRelations(final Term dnfDisjunct) {
-		final List<SolvedBinaryRelation> solvedRelations = new ArrayList<>();
-		for (final Term conjunct : SmtUtils.getConjuncts(dnfDisjunct)) {
-			solvedRelations.addAll(solveForAllSubjects(conjunct));
-		}
-		return solvedRelations;
+	@Override
+	protected NonrelationalState<Interval> getTopState() {
+		return new NonrelationalState<>();
```

Comment:
Shouldnt that return the unified top state?

Reply:
Okay, this is more complicated, since NonrelationalState is parametrized. I would just leave it like that (it is just called in case of a timeout anyway), instead of doing some unsafe casts 😉